lib/awesome_spawn.rb

```diff
@@ -64,15 +64,12 @@ def run(command, options = {})
       options[:stdin_data] = in_data
     end
 
-    output, error, status = "", "", nil
     command_line = build_command_line(command, params)
 
     output, error, status = launch(command_line, options)
-  rescue Errno::ENOENT => err
-    raise NoSuchFileError.new(err.message) if NoSuchFileError.detected?(err.message)
```

Comment:
Why was the detection logic removed?  There are cases where ENOENT could be raised, but not be related to the executable being missing. (http://stackoverflow.com/a/19902850)  In those cases we want the ENOENT to bubble out.